go/database/mpt/hasher.go

```diff
@@ -467,27 +467,27 @@ func (h ethHasher) getHash(ref *NodeReference, source NodeSource) (common.Hash,
 }
 
 // encode computes the RLP encoding of the given node. If needed, additional nodes are
-// fetched from the given manager/source for deriving the encoding. If the manager is
-// provided, write access to required nodes is obtained and dirty node information like
-// hashes and embedded flags are updated. If the manager is nil, this operation is a
-// read-only operation accepting the current hashes and embedded flags as the true value
-// even if dirty flags are set. The node and source parameter must not be nil.
-func (h ethHasher) encode(
+// fetched from the given source for deriving the encoding.
+// The node and source parameter must not be nil.
+// The result is stored in the input slice, and the slice is returned as well.
+// The reason for having the slice on the input is that the encoding of nodes
+// is a frequent operation and the slice can be reused to reduce memory allocations.
+func encode(
```

Comment:
Good idea to make this stand-alone, detached from the hasher. 

However, in this case we share the general MPT package namespace with other functions and should be more specific with the function names. Maybe `encodeToRlp` or `rlpEncode`?

Reply:
ok, used `encodeToRlp`